### DISCUSSION TITLE
How to integrate room cleaning into automations?

### DISCUSSION THREAD

**Discussion by gmp91** [PiotrMachowski/lovelace-xiaomi-vacuum-map-card#492] [Q&A]
First off, thank you @PiotrMachowski  for your work! 
I'm fairly new to HA (2 months) and installed the lovelace card a week ago.

What I want to do:
When the robot cleans a certain room at night, I want my lights to turn on for better orientation of the robot.
Also I want to automate room cleaning with automations in HA.

Any tips or ideas how to do this?

**Comment by PiotrMachowski** [comment]
> When the robot cleans a certain room at night, I want my lights to turn on for better orientation of the robot.

This case should be easily achievable by creating an automation that will turn on the lights when state of the vacuum is not "docked".

> Also I want to automate room cleaning with automations in HA.

You can figure out appropriate service calls using "Copy service call" button in card editor

**Reply by ConceptConnor** [reply]
The Copied Service Call gives me an error when trying to run the action: extra keys not allowed @ data[0]['domain'].
Here is the Service Call that does not work:
```
{
  "domain": "xiaomi_miio",
  "service": "vacuum_clean_segment",
  "serviceData": {
    "segments": [
      "8"
    ],
    "entity_id": "vacuum.dreame_p2028_f239_robot_cleaner"
  }
}
```

Is there a solution I am missing? If I start a room clean from the map it works fine.

**Reply by PiotrMachowski** [reply]
@ConceptConnor 
```yaml
  service: xiaomi_miio.vacuum_clean_segment
  data:
    segments: ["8"]
    entity_id: vacuum.dreame_p2028_f239_robot_cleaner
```
